File: app/src/main/java/br/com/ifood/tweetmood/presentation/di/AppComponent.java
package br.com.ifood.tweetmood.presentation.di;

import br.com.ifood.tweetmood.TMApplication;
import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.DaggerApplication;
import dagger.android.support.AndroidSupportInjectionModule;

/**
 * Created by uchoa on 16/06/18.
 */

@Component( modules = {
        AndroidSupportInjectionModule.class,
        AppModule.class,
        ActivityBuilder.class
})
public interface AppComponent extends AndroidInjector<DaggerApplication> {

    void inject(TMApplication app);

    @Override
    void inject(DaggerApplication instance);

    @Component.Builder
    interface Builder{
        @BindsInstance
        Builder application(TMApplication application);
        AppComponent build();
    }
}

File: app/src/main/java/com/test/ifood/twitterhumour/app/di/AppComponent.kt
package com.test.ifood.twitterhumour.app.di

import com.test.ifood.twitterhumour.app.App
import com.test.ifood.twitterhumour.datasource.di.DataSourceModule
import com.test.ifood.twitterhumour.network.di.NetworkModule
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton


@Singleton
@Component(modules = [AndroidSupportInjectionModule::class, AppModule::class, ActivityBinding::class, NetworkModule::class, DataSourceModule::class, DialogFragmentBinding::class])
interface AppComponent: AndroidInjector<App> {

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(app: App): Builder

        fun build(): AppComponent
    }
}